File: tests/test_dart/lib/services/user_service.dart
import 'package:logging/logging.dart';
import '../models/user.dart';

class UserService {
  final Logger _logger = Logger('UserService');

  void save(User user) {
    _logger.info('Saving user: ${user.name}');
  }

  List<User> findAll() {
    return [];
  }

  bool update(User user) {
    _logger.info('Updating user: ${user.name}');
    return true;
  }
}

File: tests/test_elixir/lib/my_app/services/user_service.ex
defmodule MyApp.Services.UserService do
  alias MyApp.Models.User
  require Logger

  def save(%User{} = user) do
    Logger.info("Saving user: #{user.name}")
  end

  def find_all do
    []
  end

  def update(%User{} = user) do
    Logger.info("Updating user: #{user.name}")
    true
  end
end
